Sources/SwiftProtobuf/Message+BinaryAdditions.swift

```diff
@@ -59,6 +57,34 @@ extension Message {
     return visitor.serializedSize
   }
 
+  /// Writes the message in the Protocol Buffer binary format into the given buffer.
+  ///
+  /// - Parameters:
+  ///   - buffer: An `UnsafeMutableRawBufferPointer` into which the serialized message
+  ///     will be written. The buffer should be at least `serializedDataSize()` bytes
+  ///     in size.
+  ///   - partial: If `false` (the default), this method will check
+  ///     `Message.isInitialized` before encoding to verify that all required
+  ///     fields are present. If any are missing, this method throws
+  ///     `BinaryEncodingError.missingRequiredFields`.
+  /// - Returns: The number of bytes written into the buffer.
+  /// - Throws: `BinaryEncodingError` if encoding fails.
+  @discardableResult
```

Comment:
Seems like we shouldn't add this annotation as callers should always use this value, no?